content/blog/2021-03-26-cdn-demo.md

```diff
@@ -0,0 +1,254 @@
+---
+title: "Content Delivery Network with Zenoh"
+date: 2021-03-27
+menu: "blog"
+weight: 300
+description: "27 March 2021 -- Paris."
+draft: false
+---
+
+## Why a Content Delivery Network?
+A Content Delivery Network (CDN) is a popular solution based on geographically distributed servers that work together by collecting the data to the user’s nearest node. Edge servers, located close to the end users, are the base infrastructure a CDN is built upon.
+
+CDN is increasingly important in sharing internet contents, such as images, videos, html pages, and so on, and most of the nowadays traffic is served through a CDN in order to improve users’ experience and ability to manage increasingly large media and contents.
+
+Among the several advantages the CDN brings, the main once are:
+
+- Performance and latency
+  - Less traffic on server that produces the content
+  - Less network latency
+- Resource consumption
+  - Lower overal bandwidth usage
+- Security
+  - Region-aware server
+  - DDoS prevention
+
+
+## General CDN vs. Zenoh CDN
+The architecture and the geographical coverage and presence of a CDN may differ in relation to the content and the service to serve but the basic concepts and building blocks are the following:
+- **Origin server.** The server that produces the initial content.
+- **Point of presence (POP).** Each region is served by a CDN based on a POP that can be considered as an edge server that manages the traffic for the region and guarantees the local CDN functions such as caching, security (in particular DDOS prevention), networking and content delivery. 
+- **Users.** Within a certain region, users connect to the corresponding edge server instead of the origin server.
+- When the user request data, based on monitoring and metrics, those data are trasfered from the origin server to the edge server.
+
+Here a schematic architecture of a CDN:
+
+![msg-sec](../../img/cdn.png)
+
+[**Zenoh**](https://github.com/eclipse-zenoh/zenoh) is a multi-functions protocol that provides pub/sub messaging, geo-distributed storages, queries and computations. The flexibility of Zenoh primitives allows to adopt it also in a wide range of usecases, only apparently far away from it, such as a CDN implementation.
```

Comment:
I would rephrase as follows:

[**Zenoh**](https://github.com/eclipse-zenoh/zenoh) is a protocol that unifies data in motion, data at rest and computations. It  blends traditional pub/sub with geo-distributed storages, queries and computations, while retaining a level of time and space efficiency that is well beyond any of the mainstream stacks.  Zenoh primitives allow cover  a wide range of use-cases, and as shown in this post, make a CDN implementation  rather straightforward.